src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package service.impl;
+
+import db.Storage;
+import java.util.List;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            if (!Storage.reports.containsKey(transaction.getFruit())) {
+                Storage.reports.put(transaction.getFruit(), getHowManyHaveChanged(transaction));
+            } else {
+                Storage.reports.put(transaction.getFruit(), getBalance(transaction)
+                        + getHowManyHaveChanged(transaction));
+            }
```

Comment:
Direct manipulation of the storage within the service class violates the encapsulation principle. Consider creating a separate storage service that handles all interactions with the `Storage`.